src/components/App.js

```diff
@@ -1,42 +1,83 @@
+/* eslint-disable no-unused-vars */
+import React, { useState, Component, Fragment } from 'react';
 import ButtonPannel from './ButtonPanel';
 import Display from './Display';
 import Button from './Button';
-// eslint-disable-next-line no-unused-vars
 import calculate from '../logic/calculate';
 
-function App() {
-  const buttons = [
-    'AC', 'X²', '%', '/',
-    '9', '8', '7', '*',
-    '6', '5', '4', '+',
-    '3', '2', '1', '-',
-    '0', '.', '+/-', '=',
-  ];
+const buttons = [
+  'AC', 'X²', '%', '/',
+  '9', '8', '7', '*',
+  '6', '5', '4', '+',
+  '3', '2', '1', '-',
+  '0', '.', '+/-', '=',
+];
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      total: null,
+      prev: '0',
+      next: '0',
+      operator: null,
+    };
+  }
 
-  return (
-    <>
-      <div className="App">
-        <div className="App__calculator">
-          <Display />
-          <ButtonPannel>
-            {buttons.map(button => {
-              const operators = ['+', '-', '*', '/', '%', 'X²'];
-              const isOperator = operators.includes(button);
-              const isAcBtn = button === 'AC';
-              return (
-                <Button
-                  key={button}
-                  name={button}
-                  operator={isOperator}
-                  acBtn={isAcBtn}
-                />
-              );
-            })}
-          </ButtonPannel>
+  handledisplay = (prev, next, tot, op) => {
+    let ans;
+    if (tot) {
+      ans = tot;
+    } else if (!op) {
+      ans = prev;
+    } else {
+      ans = next;
+    }
+    return ans;
+  };
```

Comment:
It would be nice to have a good naming convention.
You can write `tot` and `op` in full.